From 3e88bcb4b936241636ccb8f9fe0fc807514b8aea Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Tue, 21 Apr 2015 18:18:04 +0100
Subject: [PATCH] Initial hostname fix

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 scripts/rabbitmq-env | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/rabbitmq-env b/scripts/rabbitmq-env
index 8a1b4d6..1e326de 100755
--- a/scripts/rabbitmq-env
+++ b/scripts/rabbitmq-env
@@ -60,11 +60,11 @@ fi
 [ "x" = "x$RABBITMQ_USE_LONGNAME" ] && RABBITMQ_USE_LONGNAME=${USE_LONGNAME}
 if [ "xtrue" = "x$RABBITMQ_USE_LONGNAME" ] ; then
     RABBITMQ_NAME_TYPE=-name
-    [ "x" = "x$HOSTNAME" ] && HOSTNAME=`env hostname -f`
+    [ "x" = "x$HOSTNAME" ] && HOSTNAME=`env hostnamectl --transient`
     [ "x" = "x$NODENAME" ] && NODENAME=rabbit@${HOSTNAME}
 else
     RABBITMQ_NAME_TYPE=-sname
-    [ "x" = "x$HOSTNAME" ] && HOSTNAME=`env hostname`
+    [ "x" = "x$HOSTNAME" ] && HOSTNAME=`env hostnamectl --static`
     [ "x" = "x$NODENAME" ] && NODENAME=rabbit@${HOSTNAME%%.*}
 fi
 
-- 
1.9.1

